android.toolchain.cmake

```diff
@@ -1026,6 +1030,50 @@ if( BUILD_WITH_ANDROID_NDK )
   else()
    set( __libstl                "${__libstl}/libs/${ANDROID_NDK_ABI_NAME}/libstdc++.a" )
   endif()
+ elseif( ANDROID_STL MATCHES "c\\+\\+_shared" )
+    set( ANDROID_EXCEPTIONS       ON )
+    set( ANDROID_RTTI             ON )
+    set( ANDROID_CXX_ROOT     "${ANDROID_NDK}/sources/cxx-stl/" )
+    set( ANDROID_LLVM_ROOT    "${ANDROID_CXX_ROOT}/llvm-libc++" )
+    if( X86 ) 
```

Comment:
In order to match how the NDK works, this should probably be
if ( X86 OR X86_64 OR MIPS OR MIPS64 ) same with below on 1056.

It also looks like this has since been fixed in the NDK, so in future versions this will not be needed.
maybe
if ( (X86 OR X86_64 OR MIPS OR MIPS64) AND ANDROID_NDK_RELEASE_NUM LESS 10006 ) so that llvm-libc++-abi gets picked up when the next NDK is released.